### PR TITLE
[Form] Text Validation

### PR DESCRIPTION
**Pull request by aahlawat**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/form/Text.java [hunk 1]
```diff
@@ -1,110 +1,120 @@
...
+     * @since com.adobe.cq.wcm.core.components.models.form 13.0.0; marked <code>default</code> in 14.1.0
+     */
+    default boolean hideTitle() {
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * Returns the regex pattern to be validated.
+     *
+     * @return the regex pattern provide by user for validating form
```
Comment (abhishekagg19) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/form/Text.java]: typo - ''provided by author user"

File: content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/clientlibs/editor/.content.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:ClientLibraryFolder"
+    categories="[core.wcm.components.form.text.v2.editor]"/>
```
Comment (abhishekagg19) [inline content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/clientlibs/editor/.content.xml]: @aahlawat Do we need to update categories here?

File: content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/clientlibs/site/.content.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:ClientLibraryFolder"
+    allowProxy="{Boolean}true"
+    categories="[core.wcm.components.form.text.v2]"/>
```
Comment (abhishekagg19) [inline content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/clientlibs/site/.content.xml]: @aahlawat  Do we need to update categories here?

File: content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/clientlibs/site/js/text.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        },
+        /**
+         * A validation message to display if no input is supplied, but input is expected for the field.
+         *
+         * @type {String}
+         */
+        requiredMessage: {
+        },
+        /**
+         * A validation message to display if no input is supplied, but input is expected for the field.
```
Comment (abhishekagg19) [inline content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/clientlibs/site/js/text.js]: @aahlawat Correct the description

File: content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/_cq_dialog/.content.xml [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+                                                fieldDescription="Message displayed as tooltip when submitting the form if the value does not validate the Type chosen"
+                                                fieldLabel="Constraint Message"
+                                                name="./constraintMessage"
+                                                wrapperClass="cmp-form-text__constraintmessage"/>                                            
+                                             <regex
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                                fieldDescription="Enter custom regex pattern to vaidate value before submitting the form"
```
Comment (abhishekagg19) [inline content/src/content/jcr_root/apps/core/wcm/components/form/text/v3/text/_cq_dialog/.content.xml]: Correct typo - validate

**Comment by vladbailescu** [comment]
Merging this into a feature branch. Development has paused for this feature but will be picked up again at a later time.

**Pull request merged**